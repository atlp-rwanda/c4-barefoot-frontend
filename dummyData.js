export const user = [
  {
    "first_name":"M",
    "last_name":"JAckson",
    "email":"jackson.novelty@gmail.com",
    "occupation":"Programming",
    "username":"jacks",
    "bio":"Software Developer",
    "address":"kk 509 st",
    "language":"Kinyarwanda",
    "password":"12345678",
    "profile_picture":"https://res.cloudinary.com/mjackson/image/upload/v1609304524/Screen_Shot_2020-12-29_at_08.21.43_qe7pem.png"
  }
]

export const signupState = {
    requesting: false,
    errorOpen: false,
    erro: ""
  }
export const locationsPayload = [
    {
      id: "c6028e0d-ef88-4693-ab49-f37669891724",
      LocationName: "Kigali",
      country: "Rwanda",
      description: "The land of a thousand hills",
      link: "https://www.nationalgeographic.com/content/dam/travel/Guide-Pages/africa/downtown-kigali-rwanda.adapt.1900.1.jpg",
    }
  ]

export const accommodationsPayload = [
    {
      
        id: "0ce36391-2c08-3074-bddb-a4ea8cccbbc5",
        country: "Rwanda",
        city: "Kigali",
        title: "Marriot Hotel",
        description: "A serene environment for relaxation",
        photos: "https://cache.marriott.com/marriottassets/marriott/KGLMC/kglmc-exterior-0030-hor-feat.jpg",
    
    }
  ]

  export const accommodationState = {
    pending: false,
    accommodations: accommodationsPayload,
    error: null
  }

  export const locationState = {
    pending: false,
    locations: locationsPayload,
    error: null
  }

