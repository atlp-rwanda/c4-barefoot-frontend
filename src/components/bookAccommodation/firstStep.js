import React,{useEffect,useState} from "react";
import {Card,CardContent, CardActionArea, CardActions, CardMedia,Typography,Grid,Divider,Button,FormControlLabel,Checkbox} from '@material-ui/core';
import { Field, Form, Formik, FormikConfig,FormikValues} from 'formik';
import {getAccommodationsByLocation,selectAccommodation} from "../../redux/actions/fetchAccommodationByLocation";
import {getAccommodation,getAccommodations,getAccommodationAminity} from "../../redux/actions/fetchAccommodations";
import {getTemperature} from "../../redux/actions/getWeather";
import { connect } from 'react-redux';
import AccommodationCard from "../AccommodationCardWithReview";
import { makeStyles } from '@material-ui/core/styles';
import { Label, Place } from '@material-ui/icons'
import colors from '../colors';
import { Pagination } from '@material-ui/lab';
import Ratings from '../RatingStars';
import { Skeleton } from '@material-ui/lab';
import CloudIcon from '@material-ui/icons/Cloud';

const useStyles = makeStyles((theme) => ({
    media: {
      height: 440
    },
    checkbox:{
        display: 'block',
        position:'absolute',
        color: 'secondary',
        right: 0,
        fontSize:'25px'
    },
    cardActions:{
        display: 'flex',
        flexDirection:'column',
        alignItems:'flex-start',
    },
    separate:{
        marginLeft:theme.spacing(3),
        marginBottom:theme.spacing(3),
    },
    divider:{
        marginTop:theme.spacing(4)
    },
    separator:{
        marginBottom:theme.spacing(3),
        marginTop:theme.spacing(3),
        marginLeft:theme.spacing(2)
    },
    reviews:{
        fontSize: '14px',
        color: colors.primary100,
    },
    cardContent:{
      overflow: 'hidden'
    },
    container:{
        [theme.breakpoints.up("xs")]:{
            marginLeft:theme.spacing(3)
          },
          [theme.breakpoints.up("sm")]:{
            marginLeft:theme.spacing(9)
          }
        
    },
    btncontainer1:{
        display:"flex",
        justifyContent:"flex-start"
    },
    btncontainer2:{
        display:"flex",
        justifyContent:"flex-end"
    },
    titleText:{
      [theme.breakpoints.down('sm')]:{
        fontSize:'18px',
      }
    },
    item:{
        display:'block'
    }
  }));
function Home(props){
  const classes = useStyles();
    let temp= null;
    const [page, setPage] = useState(1);
    if(props.temp){
        temp=<Typography gutterBottom variant="h6"  className={classes.titleText} >
                <CloudIcon color="primary"/> {props.temp-273.15}                                  
            </Typography>
    }
    const display = props.accommodations.length ? 'none' : 'flex';
    const count=()=>{
        if(props.count % 6 ===0){
            return(props.count/6)
        }else{
            console.log("herre")
            return Math.trunc((props.count/6)+1)
        }
    }
    const handleChange = (event, value) => {
        props.getAccommodationsByLocation(props.accId,value)
        setPage(value)
      };
    return(
        <React.Fragment>
            <Card>
                <CardContent>
                    <Formik initialValues={{
                        from:null,
                        retrunDate:null
                    }}
                    onSubmit={values => {
                        props.nextStep();
                    }}>
                        <Form>
                        {(props.status ? 
                            <CardActionArea>
                            <Skeleton animation="wave" variant="rect" className={classes.media} />
                            <CardContent className={classes.cardContent} >
                                <Skeleton animation="wave" height={30} width="60%" />
                                <Skeleton animation="wave" height={10} width="80%" />
                            </CardContent>
                            <CardActions className={classes.cardActions}>
                                <Skeleton animation="wave" height={30} width="60%" />
                            </CardActions>
                            </CardActionArea>
                            :
                            <div >
                                <Grid container item xs={12}  className={classes.title,classes.container}  direction="column">
                                    <Typography variant="h6" style={{color: colors.primary100}} className={classes.separator}> 
                                        Choose Accommodation:
                                    </Typography>
                                    <Divider style={{width:'80%'}} variant='middle' />
                                                

                                </Grid>
                                <Grid container item xs={12} className={classes.container} direction='row'>
                                    {props.accommodations.length>0?(
                                        <Grid item>
                                        <Typography variant="subtitle1" style={{color: colors.primary100}} className={classes.separator}> 
                                            <Place color="secondary"/>{props.nation}
                                        </Typography>
                                    </Grid>
                                    ):(null)}
                                    
                                    <Grid  container item >
                                        {props.accommodations.map((accommodation) =>(
                                            <Grid item xs={8} sm={4} md={3} className={classes.insideGrid,classes.separate}>
                                                <AccommodationCard pending={props.status} accommodationn={accommodation} city={accommodation.city} {...props}  />
                                            </Grid>
                                        ))}
                                    </Grid>
                                    <Grid container item style={{display: display}} className={classes.notFound}>
                                        <Typography variant="h6" color="secondary" component="h6">No Accommodations found in this Locaton</Typography>
                                    </Grid>
                                    <Grid container item justify="center" style={{marginTop:'50px'}}>
                                    <Pagination count={count()} page={page} variant="outlined" color="primary" onChange={handleChange} />
                                    </Grid>
                                </Grid>
                                <div className={classes.btncontainer2}>
                                    <div className={classes.textCenter,classes.divider}>
                                        {props.selectedAccommodation?(<Button
                                            type='submit'
                                            variant='contained'
                                            color='primary'
                                            className={classes.button}
                                        >
                                            Next
                                        </Button>):(null)}
                                    </div>
                                </div>
                            </div>
                        )}
                        </Form>
                    </Formik>
                </CardContent>
            </Card>
        </React.Fragment>
    )
}

const mapStateToProps=state=>({
    accommodations:state.fetchAccommodations.accommodationsByLocation,
    accommodation:state.fetchAccommodations.accommodation,
    selectedAccommodation:state.fetchAccommodations.selectedAccommodation,
    count:state.fetchAccommodations.count,
    nation:state.fetchAccommodations.nation,
    accId:state.fetchAccommodations.accId,
    status:state.fetchAccommodations.pending,
    amenities:state.fetchAccommodations.amenities,
    temp:state.fetchAccommodations.temp
})
export default connect(mapStateToProps,{getAccommodationsByLocation,selectAccommodation,getAccommodation,getAccommodations,getAccommodationAminity,getTemperature}) (Home)