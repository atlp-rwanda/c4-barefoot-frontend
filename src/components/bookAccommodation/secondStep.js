import React,{useEffect,useState} from "react";
import {Card,CardContent, CardActionArea, CardActions, CardMedia,Typography,Grid,Divider,Button,FormControlLabel,Checkbox,Select,MenuItem} from '@material-ui/core';
import { Field, Form, Formik, FormikConfig,FormikValues} from 'formik';
import {getAccommodationsByLocation,selectAccommodation} from "../../redux/actions/fetchAccommodationByLocation";
import {getAccommodation,getAccommodations,getAccommodationAminity} from "../../redux/actions/fetchAccommodations";
import {getTemperature} from "../../redux/actions/getWeather";
import { connect } from 'react-redux';
import AccommodationCard from "../AccommodationCardWithReview";
import { makeStyles } from '@material-ui/core/styles';
import { Label, Place } from '@material-ui/icons'
import colors from '../colors';
import { Pagination } from '@material-ui/lab';
import Ratings from '../RatingStars';
import { Skeleton } from '@material-ui/lab';
import CloudIcon from '@material-ui/icons/Cloud';
import {convertorAction} from "../../redux/actions/convertorAction"

const useStyles = makeStyles((theme) => ({
    // root: {
    //   maxWidth: 345,
    //   height: 360,
    //   height: '100%',
    //   display: 'flex',
    //   flexDirection: 'column',
    //   justifyContent: 'space-between'
    // },
    media: {
      height: 440
    },
    checkbox:{
        display: 'block',
        position:'absolute',
        color: 'secondary',
        right: 0,
        fontSize:'25px'
    },
    cardActions:{
        display: 'flex',
        flexDirection:'column',
        alignItems:'flex-start',
    },
    btncontainer:{
        display:"flex",
        justifyContent:"space-between"
        // width:'80%'
    },
    btncontainer1:{
        //display:"flex",
        justifyContent:"flex-start"
        // width:'80%'
    },
    btncontainer2:{
        //display:"flex",
        justifyContent:"flex-end"
        // width:'80%'
    },
    separate:{
        // marginBottom:theme.spacing(3),
        marginLeft:theme.spacing(3)
    },
    divider:{
        // marginBottom:theme.spacing(3),
        marginTop:theme.spacing(4)
    },
    separator:{
        marginBottom:theme.spacing(3),
        marginTop:theme.spacing(3),
        marginLeft:theme.spacing(2)
    },
    reviews:{
        fontSize: '14px',
        color: colors.primary100,
    },
    cardContent:{
      overflow: 'hidden'
    },
    container:{
        marginLeft:theme.spacing(9),
        // width:'80%'
    },
    titleText:{
      [theme.breakpoints.down('sm')]:{
        fontSize:'18px',
      }
    },
    item:{
        display:'block'
    }
  }));
function Home(props){
    
  const classes = useStyles();
    let temp= null;
    let currency=null;
    const base={Rwanda:"RWF",Uganda:"UGX",USA:"USD",Europe:"EUR",Canada:"CND",Kenya:"KSH",Burundi:"BIF",Tanzania:"TSH"}
    const [direction, setDirection] = useState('back');
    const [value,setValue]=useState(base[props.accommodation.country]);
    const from=base[props.accommodation.country]
   
    if(props.temp){
        temp=<Typography gutterBottom variant="h6" color="textSecondary" className={classes.titleText} >
                <CloudIcon color="primary"/> {Math.round(props.temp-273.15)}                             
            </Typography>
    }
    const hanldeSelectOnchange=(e)=>{
        setValue(e.target.value)
        currency=e.target.value
        console.log(currency)
        props.convertorAction(props.accommodation.price,from,e.target.value)
     
    }
    const populateChecbox=()=>{
        let labels=[];
        let label;
        let count=0;
        if(props.amenities){
            const perm=props.amenities;
            for(const property in perm){
              label=
              <Grid item>
                 <Typography variant="subtitle1" color="textSecondary" component="p" noWrap >
                    {property}
                </Typography>
              </Grid>
              count++;
              labels.push(label);
            }
            return labels;
          }else{
            return null;
        }
    }
    return(
        <React.Fragment>
            <Card>
                <CardContent>
                    <Formik initialValues={{
                        from:null,
                        retrunDate:null
                    }}
                    onSubmit={values => {
                        direction === 'back' ? props.prevStep() : props.nextStep();
                        }}
                    >
                        <Form>
                            <div className={classes.divider} >
                            <Card className={classes.root} >
                                {(!props.accommodation ? 
                                    <CardActionArea>
                                    <Skeleton animation="wave" variant="rect" className={classes.media} />
                                    <CardContent className={classes.cardContent} >
                                        <Skeleton animation="wave" height={30} width="60%" />
                                        <Skeleton animation="wave" height={10} width="80%" />
                                    </CardContent>
                                    <CardActions className={classes.cardActions}>
                                        <Skeleton animation="wave" height={30} width="60%" />
                                    </CardActions>
                                    </CardActionArea>
                                    :
                                <>
                                <CardActionArea>
                                    
                                    <CardMedia
                                    // onClick={handleViewMore}
                                    className={classes.media}
                                    image={props.accommodation.photos}
                                    title={props.accommodation.title}
                                    />
                                </CardActionArea>
                                    <CardContent  className={classes.cardContent} >
                                        <Typography gutterBottom variant="h5" component="h2" className={classes.titleText}>
                                        {props.accommodation.title}
                                        </Typography>
                                        <Typography variant="body2" color="textSecondary" component="p" noWrap >
                                            {props.accommodation.description}
                                        </Typography>
                                        <div className={classes.divider}>
                                        <Grid container spacing={1}  direction='row' >
                                            <Grid container item xs={3} spacing={3} >
                                                    
                                                <Grid container item xs={12} sm={12} direction='column'>
                                                    <Grid item>
                                                    <Typography gutterBottom variant="h5" component="h2" className={classes.titleText} color="primary">
                                                    Location
                                                    </Typography>
                                                        <Typography gutterBottom variant="body" component="p" className={classes.titleText} color="textSecondary">
                                                        {props.accommodation.country}
                                                        </Typography>
                                                    </Grid>
                                                    <Grid item>
                                                        <Typography gutterBottom variant="subtitle1" component="p" className={classes.titleText} color="textSecondary">
                                                        {props.accommodation.city}
                                                        </Typography>
                                                    </Grid>
                                                    <Grid item>
                                                        <Typography gutterBottom variant="subtitle1" component="p" className={classes.titleText} color="textSecondary">
                                                        {props.accommodation.state}
                                                        </Typography>
                                                    </Grid>
                                                    <Grid item>
                                                        <Typography gutterBottom variant="subtitle1" component="p" className={classes.titleText} color="textSecondary">
                                                        {props.accommodation.streetAddress}
                                                        </Typography>
                                                    </Grid>
                                                </Grid>
                                            </Grid>
                                            <Grid container item xs={3}>
                                                    
                                                <Grid container item xs={12}  direction='column'>
                                                    <Typography gutterBottom variant="h5" component="h2" className={classes.titleText} color="primary">
                                                    Capacity
                                                    </Typography>
                                                    <Grid item>
                                                        <Typography variant="subtitle1" className={classes.titleText} color="textSecondary">
                                                        {props.accommodation.propertyType}
                                                        </Typography>
                                                    </Grid>
                                                    <Grid item>
                                                        <Typography gutterBottom variant="subtitle1" component="p" color="textSecondary" className={classes.titleText}>
                                                        {props.accommodation.numberOfRooms!=null?(`${props.accommodation.numberOfRooms} bedrooms`):(null)} 
                                                        </Typography>
                                                    </Grid>
                                                    <Grid item>
                                                        <Typography variant="subtitle1" component="p" className={classes.titleText} color="textSecondary">
                                                        {props.accommodation.typeOfBed?(`Type of: ${props.accommodation.typeOfBed} bedrooms`):(null)}
                                                        </Typography>
                                                    </Grid>
                                                </Grid>
                                            </Grid>
                                            <Grid container item xs={3}>
                                                    <Typography gutterBottom variant="h5" component="h2" className={classes.titleText} color="primary">
                                                Amenities
                                                <Grid container item xs={12} spacing={0} direction='column'>
                                                    {populateChecbox()}
                                                </Grid>
                                                </Typography>
                                            </Grid>
                                            <Grid item xs={3} direction='column'>
                                                <Grid item xs={12} >
                                                    <Typography gutterBottom variant="h5"  className={classes.titleText} color="primary">
                                                        Cost/night
                                                    </Typography>
                                                    <Grid container item spacing={3} xs={12}>
                                                        <Grid item>
                                                            <Typography variant="body">{props.money?(props.money[Object.keys(props.money)[0]]):(props.accommodation.price)}</Typography>
                                                        </Grid>
                                                        <Grid item >  
                                                            <Select 
                                                                labelId='select-roles'
                                                                id='roles'
                                                                value={value}
                                                                onChange={hanldeSelectOnchange}
                                                                style={{"padding-right": '9px'}}
                                                                >
                                                                <MenuItem  key="USD" name="USD" value="USD">USD</MenuItem>
                                                                <MenuItem  key="RWF" name="RWF" value="RWF">RWF</MenuItem>
                                                                <MenuItem  key="EUR" name="EUR" value="EUR">EUR</MenuItem>
                                                                <MenuItem  key="CND" name="CND" value="CND">CND</MenuItem>
                                                                <MenuItem  key="KSH" name="KSH" value="KSH">KSH</MenuItem>
                                                                <MenuItem  key="UGX" name="UGX" value="UGX">UGX</MenuItem>
                                                                <MenuItem  key="BIF" name="BIF" value="BIF">BIF</MenuItem>
                                                                <MenuItem  key="TSH" name="TSH" value="TSH">TSH</MenuItem>    
                                                            </Select>
                                                        </Grid>
                                                    </Grid>
                                                </Grid>
                                                <Grid item xs={12} spacing={4} >
                                                    <Typography gutterBottom variant="h5"  className={classes.titleText} color="primary">
                                                        Weather
                                                    </Typography>
                                                     
                                                    {temp}
                                                        
                                                   
                                                </Grid> 
                                            </Grid>
                                        </Grid>
                                    </div>
                                    </CardContent>
                                </>
                                    )}
                                    
                                </Card>
                            </div>
                            <div className={classes.divider}>
                          
                            </div>
                            <div className={classes.divider}>
                            <div className={classes.btncontainer}>
                                <div className={classes.textCenter}>
                                    <Button
                                        type='submit'
                                        
                                        variant='contained'
                                        color='primary'
                                        className={classes.button}
                                        onClick={() => {
                                        setDirection('back'); 
                                        }}
                                    >
                                        Back
                                    </Button>
                                </div>
                            <div >
                                <Button
                                    type='submit'
                                    id='backBtn'
                                    variant='contained'
                                    color='primary'
                                    className={classes.button}
                                    onClick={() => setDirection('forward')}
                                >
                                    Next
                                </Button>
                            </div>
                        </div>
                        </div>
                        </Form>
                    </Formik>
                </CardContent>
                
            </Card>
            
        </React.Fragment>
    )
}

const mapStateToProps=state=>({
    accommodations:state.fetchAccommodations.accommodationsByLocation,
    accommodation:state.fetchAccommodations.accommodation,
    money:state.convertMoney.money,
    count:state.fetchAccommodations.count,
    status:state.fetchAccommodations.pending,
    amenities:state.fetchAccommodations.amenities,
    temp:state.fetchAccommodations.temp
})
export default connect(mapStateToProps,{getAccommodationsByLocation,selectAccommodation,getAccommodation,getAccommodations,getAccommodationAminity,getTemperature,convertorAction}) (Home)