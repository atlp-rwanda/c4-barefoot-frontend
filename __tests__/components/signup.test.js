import { shallow, mount } from "enzyme";
import React from 'react';
import toJson from "enzyme-to-json";
import { FirstStep } from "../../src/components/signup/firstStep";
import { SecondStep } from "../../src/components/signup/secondStep";
import Signup from '../../src/components/signup'
import { Confirm } from "../../src/components/signup/thirdStep";
import SocialButtons from '../../src/components/signup/socialButton'
import SideDiv from "../../src/components/signup/sideDiv";
import {act} from 'react-dom/test-utils'
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";
import configureStore from 'redux-mock-store';
import { signupState} from '../../dummyData'
import thunk from "redux-thunk";

const middlewares = [thunk]
const mockStore = configureStore(middlewares);
let store;
describe('Signup', () => {
    let formData = new FormData()
    store = mockStore({
        signup: signupState,
      });
    const props = {
        formData:formData,
        setFormData: function(){},
        nextStep: function(){},
        prevStep: function(){},
        profile_picture: 'image.jpg'
    }
    describe('Signup Rendering', ()=>{
        
        it('renders correctly first step', () => {
            const wrapper = shallow(<FirstStep {...props}/>)
            expect(toJson(wrapper)).toMatchSnapshot();
        });
        it('renders correctly Signup index', () => {
            const wrapper = shallow(<Signup {...props}/>)
            expect(toJson(wrapper)).toMatchSnapshot();
        });
        it('renders correctly Third step', () => {
            const wrapper = shallow(<Provider store={store}><Confirm {...props}/></Provider>)
            expect(toJson(wrapper)).toMatchSnapshot();
        });
        it('renders side div correctly', () => {
            const wrapper = shallow(<SideDiv />)
            expect(toJson(wrapper)).toMatchSnapshot();
        });
        it('renders Social buttons correctly', () => {
            const wrapper = shallow(<SocialButtons />)            
            expect(toJson(wrapper)).toMatchSnapshot();
        });
        
    })
    describe('Functionalities', () => {
        it('should call onChange prop with input value', () => {
        
                const onSearchMock = jest.fn();
                const component = mount(<SecondStep onChange={onSearchMock} {...props} value="custom value" />);
                act( () => {   
                    component.find('#file').simulate('change');
                })
            
        });
        it('should call onSubmit on first step', () => {
        
            const onSubmit = jest.fn();
            const component = mount(<Router><FirstStep onSubmit={onSubmit} {...props} values="custom value" /></Router>);
            act( () => {   
                component.find('[type="submit"]').first().simulate('click');
            })
        
        });
        it('should call onSubmit on Second step', () => {
        
            const onSubmit = jest.fn();
            const component = mount(<Router><SecondStep onSubmit={onSubmit} {...props} values="custom value" /></Router>);
            act( () => {   
                component.find('[type="submit"]').first().simulate('click');
            })
        
        });
        it('should call onSubmit on Third step', () => {
        
            const onClick = jest.fn();
            const component = mount(<Provider store={store}><Router><Confirm onClick={onClick} {...props} values="custom value" /></Router></Provider>);
            act( () => {   
                component.find('[id="backBtn"]').first().simulate('click');
            })
        
        });
        it('should call onClick for backward on second step', async() => {
            
            const signupRequest = jest.fn()
                         
            const component = mount(<SecondStep onClick={signupRequest} {...props} />) 
            const button = component.find('#backBtn').at(0)
            await act( async() => {
                button.simulate('click');
            });
            expect(button.length).toBe(1)
            expect(signupRequest.mock.calls.length).toEqual(0);
        });
        it('should call onClick to submit form', async() => {
            
            const signupRequest = jest.fn()
                         
            const component = mount(<Provider store={store}><Confirm {...props} /></Provider>) 
            const button = component.find('[btn="submitBtn"]').first()
            await act( async() => {
                button.simulate('click');
            });
            expect(button.length).toBe(1)
            expect(signupRequest.mock.calls.length).toEqual(0);
        });
        it('submit form', async() => {
            const component = mount(<Router store={store}><FirstStep {...props} /></Router>) 
            const button = component.find('[form-data="form-1"]').first()
            await act( async() => {
                button.simulate('submit');
            });
            expect(button.length).toBe(1)
        });

    })

})
