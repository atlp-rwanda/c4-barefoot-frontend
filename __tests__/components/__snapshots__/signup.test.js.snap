// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Signup Rendering renders Social buttons correctly 1`] = `""`;

exports[`Signup Signup Rendering renders correctly Signup index 1`] = `
<div
  className="makeStyles-container-11"
>
  <SideDiv
    loading={true}
  />
  <div
    className="makeStyles-root-8"
  >
    <div
      className="makeStyles-rightSide-9"
    >
      <Styled(MuiBox)
        className="makeStyles-box-15"
      >
        <WithStyles(ForwardRef(Skeleton))
          className="makeStyles-textBox-14"
          variant="rect"
        />
      </Styled(MuiBox)>
      <Styled(MuiBox)
        className="makeStyles-stepperContainer-16"
      >
        <Styled(MuiBox)
          className="makeStyles-stepperBox-17"
        >
          <WithStyles(ForwardRef(Skeleton))
            variant="circle"
          >
            <WithStyles(ForwardRef(Avatar)) />
          </WithStyles(ForwardRef(Skeleton))>
          <WithStyles(ForwardRef(Skeleton))
            className="makeStyles-stepperText-18"
            variant="rect"
          />
          <WithStyles(ForwardRef(Skeleton))
            className="makeStyles-stepperText-18"
            variant="rect"
          />
        </Styled(MuiBox)>
        <Styled(MuiBox)
          className="makeStyles-stepperBox-17"
        >
          <WithStyles(ForwardRef(Skeleton))
            variant="circle"
          >
            <WithStyles(ForwardRef(Avatar)) />
          </WithStyles(ForwardRef(Skeleton))>
          <WithStyles(ForwardRef(Skeleton))
            className="makeStyles-stepperText-18"
            variant="rect"
          />
          <WithStyles(ForwardRef(Skeleton))
            className="makeStyles-stepperText-18"
            variant="rect"
          />
        </Styled(MuiBox)>
        <Styled(MuiBox)
          className="makeStyles-stepperBox-17"
        >
          <WithStyles(ForwardRef(Skeleton))
            variant="circle"
          >
            <WithStyles(ForwardRef(Avatar)) />
          </WithStyles(ForwardRef(Skeleton))>
          <WithStyles(ForwardRef(Skeleton))
            className="makeStyles-stepperText-18"
            variant="rect"
          />
          <WithStyles(ForwardRef(Skeleton))
            className="makeStyles-stepperText-18"
            variant="rect"
          />
        </Styled(MuiBox)>
      </Styled(MuiBox)>
      <div>
        <div>
          <WithStyles(ForwardRef(Typography))
            className="makeStyles-instructions-13"
          >
            <FirstStep
              formData={
                Object {
                  "address": "",
                  "bio": "",
                  "email": "",
                  "first_name": "",
                  "language": "",
                  "last_name": "",
                  "occupation": "",
                  "password": "",
                  "profile_picture": "",
                  "username": "",
                }
              }
              loading={true}
              nextStep={[Function]}
              setFormData={[Function]}
            />
          </WithStyles(ForwardRef(Typography))>
        </div>
      </div>
    </div>
    <SocialButtons
      loading={true}
      stepIndex={0}
    />
  </div>
</div>
`;

exports[`Signup Signup Rendering renders correctly Third step 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Confirm
    formData={FormData {}}
    nextStep={[Function]}
    prevStep={[Function]}
    profile_picture="image.jpg"
    setFormData={[Function]}
  />
</ContextProvider>
`;

exports[`Signup Signup Rendering renders correctly first step 1`] = `
<Fragment>
  <Formik
    initialValues={FormData {}}
    onSubmit={[Function]}
    validationSchema={
      ObjectSchema {
        "_blacklist": ReferenceSet {
          "list": Set {},
          "refs": Map {},
        },
        "_excludedEdges": Array [],
        "_mutate": undefined,
        "_nodes": Array [
          "password",
          "confirmPassword",
          "email",
          "username",
          "last_name",
          "first_name",
        ],
        "_sortErrors": [Function],
        "_typeError": [Function],
        "_whitelist": ReferenceSet {
          "list": Set {},
          "refs": Map {},
        },
        "conditions": Array [],
        "deps": Array [],
        "exclusiveTests": Object {},
        "fields": Object {
          "confirmPassword": StringSchema {
            "_blacklist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_blacklistError": undefined,
            "_typeError": [Function],
            "_whitelist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_whitelistError": undefined,
            "conditions": Array [
              Condition {
                "fn": [Function],
                "refs": Array [
                  Reference {
                    "getter": [Function],
                    "isContext": false,
                    "isSibling": true,
                    "isValue": false,
                    "key": "password",
                    "map": undefined,
                    "path": "password",
                  },
                ],
              },
            ],
            "deps": Array [
              "password",
            ],
            "exclusiveTests": Object {
              "required": true,
            },
            "spec": Object {
              "abortEarly": true,
              "label": undefined,
              "meta": undefined,
              "nullable": false,
              "presence": "required",
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": Array [
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
          "email": StringSchema {
            "_blacklist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_blacklistError": undefined,
            "_mutate": undefined,
            "_typeError": [Function],
            "_whitelist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_whitelistError": undefined,
            "conditions": Array [],
            "deps": Array [],
            "exclusiveTests": Object {
              "email": false,
              "required": true,
            },
            "spec": Object {
              "abortEarly": true,
              "label": undefined,
              "meta": undefined,
              "nullable": false,
              "presence": "required",
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": Array [
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
          "first_name": StringSchema {
            "_blacklist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_blacklistError": undefined,
            "_typeError": [Function],
            "_whitelist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_whitelistError": undefined,
            "conditions": Array [],
            "deps": Array [],
            "exclusiveTests": Object {
              "max": true,
              "required": true,
            },
            "spec": Object {
              "abortEarly": true,
              "label": undefined,
              "meta": undefined,
              "nullable": false,
              "presence": "required",
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": Array [
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
          "last_name": StringSchema {
            "_blacklist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_blacklistError": undefined,
            "_typeError": [Function],
            "_whitelist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_whitelistError": undefined,
            "conditions": Array [],
            "deps": Array [],
            "exclusiveTests": Object {
              "max": true,
              "required": true,
            },
            "spec": Object {
              "abortEarly": true,
              "label": undefined,
              "meta": undefined,
              "nullable": false,
              "presence": "required",
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": Array [
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
          "password": StringSchema {
            "_blacklist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_blacklistError": undefined,
            "_typeError": [Function],
            "_whitelist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_whitelistError": undefined,
            "conditions": Array [],
            "deps": Array [],
            "exclusiveTests": Object {
              "min": true,
              "required": true,
            },
            "spec": Object {
              "abortEarly": true,
              "label": undefined,
              "meta": undefined,
              "nullable": false,
              "presence": "required",
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": Array [
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
          "username": StringSchema {
            "_blacklist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_blacklistError": undefined,
            "_typeError": [Function],
            "_whitelist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_whitelistError": undefined,
            "conditions": Array [],
            "deps": Array [],
            "exclusiveTests": Object {
              "max": true,
              "required": true,
            },
            "spec": Object {
              "abortEarly": true,
              "label": undefined,
              "meta": undefined,
              "nullable": false,
              "presence": "required",
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": Array [
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
        },
        "spec": Object {
          "abortEarly": true,
          "label": undefined,
          "meta": undefined,
          "nullable": false,
          "presence": "optional",
          "recursive": true,
          "strict": false,
          "strip": false,
        },
        "tests": Array [],
        "transforms": Array [
          [Function],
        ],
        "type": "object",
      }
    }
  >
    <Component />
  </Formik>
</Fragment>
`;

exports[`Signup Signup Rendering renders side div correctly 1`] = `
<div
  className="makeStyles-sideNav-19"
>
  <div
    className="makeStyles-sideText-23"
  >
    <p>
      Please create account to get 
    </p>
    <p>
      started with Barefoot Nomard.
    </p>
  </div>
  <div
    className="makeStyles-img-21"
  >
    <img
      alt="Barefoot Nomards"
      className="makeStyles-locationImg-20"
      src="https://res.cloudinary.com/barefoot-nomad-app/image/upload/v1609407601/images/location_goruvq.png"
    />
  </div>
</div>
`;
